hyperspy/_signals/eds_tem.py

```diff
@@ -353,47 +381,133 @@ def quantification(self,
         --------
         vacuum_mask
         """
-        if self.axes_manager.navigation_size == 0:
-            navigation_mask = None
-        elif isinstance(navigation_mask, float):
+        if isinstance(navigation_mask, float):
             navigation_mask = self.vacuum_mask(navigation_mask, closing).data
         elif navigation_mask is not None:
             navigation_mask = navigation_mask.data
 
+        xray_lines = [intensity.metadata.Sample.xray_lines[0] for intensity in intensities]
+        it = 0
+        if absorption_correction:
+            pbar = progressbar(total=max_iterations+1)
+
         composition = utils.stack(intensities, lazy=False)
+        if take_off_angle == 'auto':
+            toa = self.get_take_off_angle()
+        else:
+            toa = take_off_angle
+
+        #determining illumination area for cross sections quantification.
+        if method == 'cross_section':
+            if probe_area == 'auto':
+                parameters = self.metadata.Acquisition_instrument.TEM
+                if probe_area in parameters:
+                    probe_area = parameters.TEM.probe_area
+                else:
+                    probe_area = self._get_probe_area(navigation_axes=None)
```

Comment:
In case of `navigation dimension > 2`, this is going to lead to a confusing error message asking the user to specify the `navigation_axes` parameter but the `quantification` function does not take this argument. No sure, what is the best, maybe tweak the error message in `_get_probe_area`?